Sources/Testing/Test.swift

```diff
@@ -128,6 +128,23 @@ public struct Test: Sendable {
     }
   }
 
+  /// Equivalent to ``testCases``, but without requiring that the test cases be
+  /// evaluated first.
+  ///
+  /// Most callers should not use this property and should prefer ``testCases``
+  /// since it will help catch logic errors in the testing library. Use this
+  /// property if you are interested in the test's test cases, but the test has
+  /// not been evaluated by an instance of ``Runner/Plan`` (e.g. if you are
+  /// implementing `swift test list`.)
+  var uncheckedTestCases: (some Sequence<Test.Case>)? {
```

Comment:
What's the connection to this PR of this change?

Reply:
Crashes without it :)

Reply:
I do think it would be useful to evaluate test cases when listing tests by default (perhaps with an opt-out). That can be considered out of scope for this PR, although I also would't object if you instead changed this to evaluate them as a way to fix this crash.

Reply:
Let's consider that a "future direction" for now as the evaluated test cases aren't formally part of the JSON spec yet. (This part of the change doesn't affect snapshots.)

Reply:
Sounds good